src/creation/mod.rs

```diff
@@ -106,12 +106,15 @@ pub async fn run<H: Hasher>(
         mut incoming_parents,
         outgoing_units,
     } = io;
-    let starting_round = match starting_round.await {
-        Ok(round) => round,
-        Err(e) => {
-            error!(target: "AlephBFT-creator", "Starting round not provided: {}", e);
-            return;
-        }
+    let starting_round = futures::select! {
+        round =  starting_round => match round {
+            Ok(round) => round,
+            Err(e) => {
+                error!(target: "AlephBFT-creator", "Starting round not provided: {}", e);
+                return;
+            }
+        },
+        _ = &mut exit => return,
     };
     debug!(target: "AlephBFT-creator", "Creator starting from round {}", starting_round);
```

Comment:
Now sure why my IDE says type of `starting_round` at this line is `()`. Had it been `()`, it would not have compiled. On the other hand I'm a bit confused here, since we do assign to `starting_round` result of `select!`, which indeed returns `()`, as all its branches must return the same, and second branch return type is `()`

![image](https://user-images.githubusercontent.com/3909333/149347576-b771e8a3-ce67-4e6d-8824-c34b5719a117.png)

Reply:
Ah my bad, `return` has _never_ type or `!` : https://doc.rust-lang.org/nightly/std/primitive.never.html
All good then. 

>! will coerce into any other type.

Reply:
The second branch return type is actually `bot`, since it returns out of the function completely.

Your IDE might be confused by the `futures` macro? I know it used to be a problem with my vim plugin, but it got better.